internal/handlers/logs.go

```diff
@@ -135,13 +139,96 @@ func (p *Log) PatchLog(ctx *fiber.Ctx) error {
 func (p *Log) DeleteLog(ctx *fiber.Ctx) error {
 	var log models.Log
 
-	if err := p.db.Delete(&log, ctx.Params("id")).Error; err != nil {
-		if errors.Is(err, gorm.ErrRecordNotFound) {
-			return common.Error(fiber.StatusNotFound, "can't delete Log", "Log was not found")
-		}
+	result := p.db.Delete(&log, "id = ?", ctx.Params("id"))
 
+	if result.Error != nil {
 		return common.Error(fiber.StatusInternalServerError, "can't delete Log", "db error")
 	}
 
+	if result.RowsAffected == 0 {
+		return common.Error(fiber.StatusNotFound, "can't delete Log", "Log was not found")
+	}
+
 	return ctx.SendStatus(fiber.StatusNoContent)
 }
+
+// GetSoftwareLogs gets the logs associated to a Software with the given ID and returns any error encountered.
+func (p *Log) GetSoftwareLogs(ctx *fiber.Ctx) error {
+	var logs []models.Log
+
+	software := models.Software{}
+
+	if err := p.db.First(&software, "id = ?", ctx.Params("id")).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return common.Error(fiber.StatusNotFound, "can't get Software", "Software was not found")
+		}
+
+		return common.Error(
+			fiber.StatusInternalServerError,
+			"can't get Software",
+			fiber.ErrInternalServerError.Message,
+		)
+	}
+
+	stmt := p.db.Where("entity_id = ? AND entity_type = 'Software'", software.ID)
+
+	paginator := general.NewPaginator(ctx)
+
+	result, cursor, err := paginator.Paginate(stmt, &logs)
+	if err != nil {
+		return common.Error(
+			fiber.StatusBadRequest,
+			"can't get Software",
+			"wrong cursor format in page[after] or page[before]",
+		)
+	}
+
+	if result.Error != nil {
+		return common.Error(
+			fiber.StatusInternalServerError,
+			"can't get Software",
+			fiber.ErrInternalServerError.Message,
+		)
+	}
+
+	return ctx.JSON(fiber.Map{"data": &logs, "links": general.PaginationLinks(cursor)})
+}
+
+// PostSoftwareLog creates a new log associated to a Software with the given ID and returns any error encountered.
+func (p *Log) PostSoftwareLog(ctx *fiber.Ctx) error {
+	logReq := new(requests.Log)
+
+	if err := ctx.BodyParser(&logReq); err != nil {
+		return common.Error(fiber.StatusBadRequest, "can't create Log", "invalid json")
+	}
+
+	if err := common.ValidateStruct(*logReq); err != nil {
+		return common.Error(fiber.StatusUnprocessableEntity, "can't create Log", "invalid format")
+	}
+
+	software := models.Software{}
+	if err := p.db.First(&software, "id = ?", ctx.Params("id")).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return common.Error(fiber.StatusNotFound, "can't get Software", "Software was not found")
+		}
+
+		return common.Error(
+			fiber.StatusInternalServerError,
+			"can't get Software",
+			fiber.ErrInternalServerError.Message,
+		)
+	}
+
+	log := models.Log{
+		ID:         utils.UUIDv4(),
+		Message:    logReq.Message,
+		EntityID:   software.ID,
+		EntityType: "Software",
```

Comment:
Same here

Reply:
That's what I didn't like: having to explicitly specify the type as a string, when we already have a `Software` struct and the ORM knows about the `Logs` <-> `Software` relation